R/ncdf.R

```diff
@@ -139,17 +166,37 @@ read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(
   dimensions <- .get_nc_time(dimensions, make_time,
                             coord_var, rep_var, meta)
 
-  # Make initial response data
-  ret = st_stars(out_data, dimensions)
+  if (is.list(out_data[[1]])) {
+    
+    ret <- as.list(rep(x, length(out_data)))
+    names(ret) <- names(out_data)
+    
+    # this is a proxy
+    ret <- st_stars_proxy(ret, dimensions, NA_value = NA_real_, resolutions = NULL)
+
+    ret <- put_nc_request(ret, out_data)
```

Comment:
This puts the unique nc_request sub-list into an attribute of each of the return variables.